packages/ui/src/models/visualization/flows/camel-route-visual-entity.ts

```diff
@@ -48,6 +49,19 @@ export class CamelRouteVisualEntity implements BaseVisualCamelEntity {
     return visualComponentSchema;
   }
 
+  getIconName(path: string | undefined): string | undefined {
+    if (!path) return undefined;
+    const componentModel = get(this.route, path);
+    const iconName = CamelComponentSchemaService.getIconName(
+      CamelComponentSchemaService.getCamelComponentLookup(path, componentModel),
+    );
+    return iconName;
+  }
+
+  setIconData(path: string | undefined): string | undefined {
+    return NodeIconResolver.getIcon(this.getIconName(path));
```

Comment:
[curious-question] Is this method doing the intended thing? It seems suspicious a `set` method that doesn't contain an assignment. :smiley: